Modules/_sqlite/connection.c

```diff
@@ -739,24 +739,32 @@ _pysqlite_set_result(sqlite3_context* context, PyObject* py_val)
         if (str == NULL) {
             return -1;
         }
-        if (sz > INT_MAX) {
+#if SIZEOF_SIZE_T > 8
+        if (sz > 0xffffffffffffffff) {
             PyErr_SetString(PyExc_OverflowError,
-                            "string is longer than INT_MAX bytes");
+                            "Error setting result value: "
+                            "string is longer than 2**64-1 bytes");
             return -1;
         }
-        sqlite3_result_text(context, str, (int)sz, SQLITE_TRANSIENT);
+#endif
```

Comment:
Hm, the SQLite API consistently uses unsigned `uint_64` values, and the Python C APIs (involved here) consistently used signed `ssize_t` values.

- if `ssize_t` is less than 8 bytes, we're ok; it will fit in `uint_64` anyway
- if `ssize_t` is 8 bytes or larger, we must check if the value is larger than `2**63-1`.

Reply:
See also the code in `_pysqlite_long_as_int64` (Modules/_sqlite/util.c).

Reply:
Thanks; I didn't have time to look at this since yesterday. Maybe I'll find time to return to it later today. Thanks for chiming in; these C data type issues are hard to get straight!

Reply:
I changed it to your former suggestion.